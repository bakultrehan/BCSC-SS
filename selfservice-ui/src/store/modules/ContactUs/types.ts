export interface TestAccountState {
  contactDetails?: any;
  isLoading: boolean;
  successStatus?: boolean;
  errorStatus?: boolean;
}
