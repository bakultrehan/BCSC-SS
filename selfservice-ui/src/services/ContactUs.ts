import axios from '@/lib/axios';
import { ContactDetailsModel } from '@/models/ContactDetailsModel';
import { CONTACTUS_URL } from '@/config/api-endpoints';
export class ContactUs {
  public static async addContactUs(contactDetails: ContactDetailsModel) {
    return await axios.post(CONTACTUS_URL, contactDetails);
  }
}
