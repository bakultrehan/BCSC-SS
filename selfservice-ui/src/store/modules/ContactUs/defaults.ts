const contactDetailsSample = () => {
  return {
    firstName: '',
    lastName: '',
    email: '',
    description: '',
  };
};

export { contactDetailsSample };
