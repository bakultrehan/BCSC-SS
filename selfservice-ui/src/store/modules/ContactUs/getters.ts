import { GetterTree } from 'vuex';
import { TestAccountState } from './types';
import { RootState } from '../../types';

/**
 * Getters Shared
 */
export const getters: GetterTree<TestAccountState, RootState> = {
  /**
   * isLoading  status
   * @param {*} state
   * @returns {boolean} - loading status
   */
  isLoading(state) {
    return state.isLoading;
  },
  getChangeStatus(state) {
    return {
      successStatus: state.successStatus,
      errorStatus: state.errorStatus,
    };
  },
};
