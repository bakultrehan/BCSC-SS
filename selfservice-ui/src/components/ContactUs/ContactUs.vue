/** 
* 
ContactUs component 
*/

<template>
  <v-card class="mx-auto outer-card">
    <v-form ref="form" v-model="valid">
      <h1 class="bc-page-title-h1">{{ $t('contactUs.pagetitle') }}</h1>
      <div class="flex-grow-1"></div>

      <v-container>
        <v-row class="mx-4">
          <v-col cols="12" flat>
            <v-card flat>
              <div class="text-left" v-html="$t('contactUs.pageinfo')"></div>
            </v-card>
          </v-col>
          <v-col class="col-12" v-if="errorStatus || successStatus">
            <Alert
              type="error"
              v-if="errorStatus"
              class="alert-top text-left"
            >{{ $t('contactUs.errorMessage') }}</Alert>
            <Alert
              type="success"
              class="alert-top text-left"
              v-if="successStatus"
            >{{ $t('contactUs.successMessage') }}</Alert>
          </v-col>
          <v-col cols="8">
            <Input
              v-model="contactDetails.firstName"
              :label="$t('contactUs.firstName')"
              type="text"
              :rules="[rules.required,  rules.maxLength(500)]"
              data-test-id="input-firstname"
            />

            <Input
              v-model="contactDetails.lastName"
              :label="$t('contactUs.lastName')"
              type="text"
              :rules="[rules.required, rules.maxLength(500)]"
              data-test-id="input-lastname"
            />
            <Input
              v-model="contactDetails.email"
              :label="$t('contactUs.emailAddress')"
              type="text"
              :rules="[rules.required, rules.email, rules.maxLength(500)]"
              data-test-id="input-email"
            />
          </v-col>
          <v-col cols="12" flat>
            <v-card flat>
              <TextArea
                v-model="contactDetails.description"
                :label="$t('contactUs.description')"
                type="text"
                outlined
                rows="10"
                name="text-contact-us-description"
                id="text-contact-us-description"
                data-test-id="text-contact-us-description"
                :rules="[rules.required]"
              />
            </v-card>
          </v-col>
        </v-row>

        <v-col cols="12">
          <v-card flat>
            <v-divider></v-divider>
            <v-card-actions>
              <v-spacer></v-spacer>

              <Button
                :disabled="!valid"
                class="white--text btn-contact-us ml-6"
                depressed
                @click="submitContactMessage"
                name="btn-contact-us"
                data-test-id="btn-contact-us"
              >{{ $t('contactUs.btnSend') }}</Button>
            </v-card-actions>
          </v-card>
        </v-col>
      </v-container>
    </v-form>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import validationRules from '@/config/validationRules';

import Button from '@/Atomic/Button/Button.vue';
import TextArea from '@/Atomic/TextArea/TextArea.vue';
import Alert from '@/Atomic/Alert/Alert.vue';
import Input from '@/Atomic/Input/Input.vue';

import { contactDetailsSample } from '@/store/modules/ContactUs/defaults';

const ContactUsModule = namespace('ContactUsModule');

@Component({
  components: {
    Button,
    TextArea,
    Alert,
    Input
  }
})
export default class ContactUs extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @ContactUsModule.Action('addContactUs')
  public addContactUs!: any;
  @ContactUsModule.Getter('isLoading')
  public isLoading!: boolean;
  @ContactUsModule.Getter('getChangeStatus')
  public getChangeStatus!: boolean;

  private rules = validationRules;

  private contactDetails: any;
  private valid: boolean = false;
  private successStatus: boolean = false;
  private errorStatus: boolean = false;

  public constructor() {
    super();
    this.contactDetails = contactDetailsSample();
  }

  @Watch('getChangeStatus')
  private ongetChangeStatusChanged(val: any) {
    const { successStatus, errorStatus } = val;
    this.successStatus = successStatus;
    this.errorStatus = errorStatus;
    if (successStatus) {
      (this.$refs.form as Vue & {
        reset: () => any;
      }).reset();
      this.$vuetify.goTo(0, {
        duration: 1000,
        easing: 'easeInOutCubic'
      });
    }
  }

  private submitContactMessage() {
    this.addContactUs({
      contactDetails: this.contactDetails
    });
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
</style>
