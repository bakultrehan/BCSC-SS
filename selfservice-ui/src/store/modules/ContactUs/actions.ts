import { ActionTree } from 'vuex';
import { TestAccountState } from './types';
import { RootState } from '../../types';
import { ContactUs } from '@/services/ContactUs';

/**
 * contact us Actions
 *
 */
export const actions: ActionTree<TestAccountState, RootState> = {
  /**
   * @param  {} {commit}
   * @param  {} data contact us data
   */
  async addContactUs({ commit, dispatch }, data: any) {
    commit('SET_LOADING', true);
    dispatch('clearStatus');

    try {
      const { contactDetails } = data;
      await ContactUs.addContactUs({ ...contactDetails });
      commit('SET_SUCESS', true);
      commit('SET_ERROR', false);
      commit('SET_LOADING', false);
    } catch {
      commit('SET_ERROR', true);
      commit('SET_SUCESS', false);
      commit('SET_LOADING', false);
    }
  },
  /**
   * clear message
   * @param {*} { commit }
   */
  async clearStatus({ commit }) {
    commit('SET_SUCESS', false);
    commit('SET_ERROR', false);
    commit('SET_LOADING', false);
  },
};
