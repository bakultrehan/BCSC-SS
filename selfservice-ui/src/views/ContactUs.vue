<template>
  <v-card flat>
    <ContactUs />
  </v-card>
</template>

<script>
// @ is an alias to /src
import { Component, Vue } from 'vue-property-decorator';
import ContactUs from '@/components/ContactUs/ContactUs.vue';

@Component({
  components: {
    ContactUs
  }
})
// Home goes here
export default class ContactUsPage extends Vue {}
</script>
