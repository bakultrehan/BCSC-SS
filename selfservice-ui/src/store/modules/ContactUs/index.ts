import { Module } from 'vuex';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { TestAccountState } from './types';
import { RootState } from '../../types';
import { contactDetailsSample } from './defaults';

/**
 * Sharedmodule state
 */
export const state: TestAccountState = {
  contactDetails: contactDetailsSample(),
  isLoading: false,
  successStatus: false,
  errorStatus: false,
};

const namespaced: boolean = true;

const SharedModule: Module<TestAccountState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations,
};

export default SharedModule;
