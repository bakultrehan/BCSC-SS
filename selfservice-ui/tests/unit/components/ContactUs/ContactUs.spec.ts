import { mount } from '@vue/test-utils';
import ContactUs from '@/components/ContactUs/ContactUs.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import Vue from 'vue';
let vuetify: any;
vuetify = new Vuetify();

describe('ContactUs.vue', () => {
  const store = new Vuex.Store({
    modules: {
      ContactUsModule: {
        namespaced: true,
        state: {},
        getters: {
          getChangeStatus: jest.fn(),
          isLoading: jest.fn(),
        },
        actions: {
          addContactUs: jest.fn(),
        },
      },
    },
  });

  const mountFunction = (options: any) => {
    return mount(ContactUs, {
      store,
      vuetify,
      sync: false,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options,
    });
  };

  it('renders props when passed with gettors', () => {
    const ContactUsPage = mountFunction({});
    expect(ContactUsPage.element).toMatchSnapshot();
  });

  // it('renders data on conact submit click ', async () => {
  //   const ContactUsPage = mountFunction({});
  //   const submitContactMessage = jest.fn();

  //   ContactUsPage.setData({
  //     contactDetails: {
  //       firstName: 'fname',
  //       lastName: 'lname',
  //       email: 'email@email.com',
  //       description: 'test',
  //     },
  //   });

  //   ContactUsPage.setData({
  //     valid: true,
  //   });

  //   await Vue.nextTick();
  //   const button = ContactUsPage.find('.btn-contact-us');
  //   ContactUsPage.vm.$on('action-btn:clicked', submitContactMessage);
  //   button.trigger('click');
  //   expect(ContactUsPage.element).toMatchSnapshot();
  // });
});
