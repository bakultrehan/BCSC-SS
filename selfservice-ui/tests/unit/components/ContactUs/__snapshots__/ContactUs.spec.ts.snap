// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs.vue renders props on submit click 1`] = `
<div
  class="mx-auto outer-card v-card v-sheet theme--light"
>
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <h1
      class="bc-page-title-h1"
    >
      
    </h1>
     
    <div
      class="flex-grow-1"
    />
     
    <div
      class="container"
    >
      <div
        class="row mx-4"
      >
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="text-left"
            />
          </div>
        </div>
         
        <!---->
         
        <div
          class="col col-8"
        >
          <div
            class="bc-form-text"
            data-test-id="input-firstname"
            rules="v => !!v || 'This field is required',v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <div
              class="text-left my-1"
            >
              
     
              <!---->
            </div>
             
            <!---->
             
            <div
              class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-firstname"
                      id="input-36"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-lastname"
            rules="v => !!v || 'This field is required',v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <div
              class="text-left my-1"
            >
              
     
              <!---->
            </div>
             
            <!---->
             
            <div
              class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-lastname"
                      id="input-40"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-email"
            rules="v => !!v || 'This field is required',v => /^\\\\w+([.-]?\\\\w+)*@\\\\w+([.-]?\\\\w+)*(\\\\.\\\\w{2,10})+$/.test(v) || 'E-mail must be valid',v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <div
              class="text-left my-1"
            >
              
     
              <!---->
            </div>
             
            <!---->
             
            <div
              class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-email"
                      id="input-44"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="bc-form-text"
              data-test-id="text-contact-us-description"
              id="text-contact-us-description"
              name="text-contact-us-description"
              outlined=""
              rows="10"
              rules="v => !!v || 'This field is required'"
            >
              <div
                class="text-left my-1"
              >
                
              </div>
               
              <!---->
               
              <div
                class="v-input v-textarea v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <fieldset
                      aria-hidden="true"
                    >
                      <legend>
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                    <div
                      class="v-text-field__slot"
                    >
                      <textarea
                        data-test-id="text-contact-us-description"
                        id="text-contact-us-description"
                        name="text-contact-us-description"
                        rows="10"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <span
                        class="v-messages__wrapper"
                        name="message-transition"
                        tag="div"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="col col-12"
      >
        <div
          class="v-card v-card--flat v-sheet theme--light"
        >
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--light"
            role="separator"
          />
           
          <div
            class="v-card__actions"
          >
            <div
              class="spacer"
            />
             
            <button
              class="btn white--text btn-contact-us ml-6 v-btn v-btn--depressed theme--light v-size--default primary normal"
              data-test-id="btn-contact-us"
              depressed=""
              name="btn-contact-us"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`ContactUs.vue renders props when passed with gettors 1`] = `
<div
  class="mx-auto outer-card v-card v-sheet theme--light"
>
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <h1
      class="bc-page-title-h1"
    >
      
    </h1>
     
    <div
      class="flex-grow-1"
    />
     
    <div
      class="container"
    >
      <div
        class="row mx-4"
      >
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="text-left"
            />
          </div>
        </div>
         
        <!---->
         
        <div
          class="col col-8"
        >
          <div
            class="bc-form-text"
            data-test-id="input-firstname"
            rules="v => !!v || 'This field is required',v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <div
              class="text-left my-1"
            >
              
     
              <!---->
            </div>
             
            <!---->
             
            <div
              class="v-input input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-firstname"
                      id="input-9"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-lastname"
            rules="v => !!v || 'This field is required',v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <div
              class="text-left my-1"
            >
              
     
              <!---->
            </div>
             
            <!---->
             
            <div
              class="v-input input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-lastname"
                      id="input-13"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-email"
            rules="v => !!v || 'This field is required',v => /^\\\\w+([.-]?\\\\w+)*@\\\\w+([.-]?\\\\w+)*(\\\\.\\\\w{2,10})+$/.test(v) || 'E-mail must be valid',v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <div
              class="text-left my-1"
            >
              
     
              <!---->
            </div>
             
            <!---->
             
            <div
              class="v-input input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-email"
                      id="input-17"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="bc-form-text"
              data-test-id="text-contact-us-description"
              id="text-contact-us-description"
              name="text-contact-us-description"
              outlined=""
              rows="10"
              rules="v => !!v || 'This field is required'"
            >
              <div
                class="text-left my-1"
              >
                
              </div>
               
              <!---->
               
              <div
                class="v-input v-textarea theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <fieldset
                      aria-hidden="true"
                    >
                      <legend>
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                    <div
                      class="v-text-field__slot"
                    >
                      <textarea
                        data-test-id="text-contact-us-description"
                        id="text-contact-us-description"
                        name="text-contact-us-description"
                        rows="10"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <span
                        class="v-messages__wrapper"
                        name="message-transition"
                        tag="div"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="col col-12"
      >
        <div
          class="v-card v-card--flat v-sheet theme--light"
        >
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--light"
            role="separator"
          />
           
          <div
            class="v-card__actions"
          >
            <div
              class="spacer"
            />
             
            <button
              aria-disabled="true"
              class="btn white--text btn-contact-us ml-6 v-btn v-btn--depressed v-btn--disabled theme--light v-size--default primary normal disabled"
              data-test-id="btn-contact-us"
              depressed=""
              disabled="disabled"
              name="btn-contact-us"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
